src/components/services/Services.tsx

```diff
@@ -123,6 +127,60 @@ export function Services(props: Props) {
     'Services 2'
   )
 
+  // Subscribe to Android Power Saver state, and show a warning only if it
+  // changes from off to on:
+  useAsyncEffect(
+    async () => {
+      let previousState = false
+      let isWarningShowing = false
+
+      const showBatterySaverWarning = async () => {
+        if (isWarningShowing) return
+
+        isWarningShowing = true
+        await Airship.show(bridge => <AlertDropdown bridge={bridge} message={lstrings.warning_battery_saver} warning persistent />)
+        isWarningShowing = false
+      }
+
+      const checkPowerSavingMode = async () => {
+        // This method is only available for Android
+        if (powerSavingOn != null) {
+          try {
+            previousState = await powerSavingOn()
+            if (previousState) {
+              // Show warning if power-saving mode is on initially
+              await showBatterySaverWarning()
+            }
+          } catch (error) {
+            showDevError(`Error checking power-saving mode: ${String(error)}`)
+          }
+        }
+      }
+
+      const handlePowerSavingModeChanged = async (state: boolean) => {
+        if (state && !previousState) {
```

Comment:
No need to have a `previousState`. `showBatterySaverWarning` already checks if it's showing and early exits.

Reply:
Yes there's a need for a 'previousState' 

`showBatterySaverWarning` can check if the modal is showing and early exits, yes. 

This statement is saying only fire when the powerSaving mode changes specifically from off to on.

Two state mechanisms exist here: 
1. Whether the modal is showing (agnostic of the power saving state)
2. Only detect changes for power saving mode from off to on